src/components/include/utils/threads/thread.h

```diff
@@ -76,36 +76,67 @@ typedef pthread_t PlatformThreadHandle;
  * printf("ok!\n");
  */
 
-class Thread;
-void enqueue_to_join(Thread* thread);
-
 Thread* CreateThread(const char* name, ThreadDelegate* delegate);
 void DeleteThread(Thread* thread);
 
 class Thread {
- private:
-  const std::string name_;
-  // Should be locked to protect delegate_ value
-  sync_primitives::Lock delegate_lock_;
-  ThreadDelegate* delegate_;
-  PlatformThreadHandle handle_;
-  ThreadOptions thread_options_;
-  // Should be locked to protect isThreadRunning_ and thread_created_ values
-  sync_primitives::Lock state_lock_;
-  volatile unsigned int isThreadRunning_;
-  volatile bool stopped_;
-  volatile bool finalized_;
-  bool thread_created_;
-  // Signalled when Thread::start() is called
-  sync_primitives::ConditionalVariable run_cond_;
-
  public:
-  static int count;
+  /**
+   * @brief ThreadCommand is used to command the thread
+   * kThreadCommandNone - no command, used to indicate that there is no pending
+   * command
+   * kThreadCommandRun - commands thread to run (do another iteration)
+   * kThreadCommandFinalize - informs thread that must exit
+   */
+  enum ThreadCommand {
+    kThreadCommandNone,  // must be first
+    kThreadCommandRun,
+    kThreadCommandFinalize  // keep last
+    // in case of new commands - update/check threadFunc()
+  };
+
+  /**
+   * @brief ThreadState informs outside world about its state
+   * kThreadStateError - if pthread_create returned an error
+   * kThreadStateNone - there is no thread at all
+   * kThreadStateIdle - the thread is in state idle
+   * kThreadStateRunning - thread is in state running (executing delegates
+   * threadMain())
+   * kThreadStateCompleted - thread completed
+   */
+  enum ThreadState {
+    kThreadStateError = -1,
+    kThreadStateNone,
+    kThreadStateIdle,
+    kThreadStateRunning,
+    kThreadStateCompleted
+  };
+
+  /**
+   * @brief ThreadStopOption
+   * kThreadStopTask - executing delegates exitThreadMain and waiting finish
+   * threadMain kThreadStopSelfSoft - executing StopTask and waiting finish
```

Comment:
```suggestion
   * kThreadStopTask - executing delegates exitThreadMain and waiting for finish
   * threadMain kThreadStopSelfSoft - executing StopTask and waiting for finish
```

Reply:
@LitvinenkoIra 
Thanks, fixed in:https://github.com/LuxoftSDL/sdl_core/pull/50/commits/978d64042c2c4f5cb23d54890ab908e95e53f475